src/software/amazon/ion/impl/lite/IonContainerLite.java

```diff
@@ -62,7 +62,15 @@ protected IonContainerLite(ContainerlessContext context, boolean isNull)
 
                 IonValueLite copy = child.clone(childContext);
                 if (isStruct) {
-                    copy.setFieldNameSymbol(child.getKnownFieldNameSymbol());
+                    if(child.getFieldName() == null) {
+                        // when name is null it could be a sid 0 so we need to perform the full symbol token lookup.
+                        // this is expensive so only do it when necessary
+                        copy.setFieldNameSymbol(child.getKnownFieldNameSymbol());
```

Comment:
Lets add a TODO and link to a Github Issue here to add the `SymbolTokenProvider` based solution.